oqupy/bath_dynamics.py

```diff
@@ -0,0 +1,310 @@
+        last_time = self._process_tensor.times[-1]
+        re_kernel,im_kernel = self._calc_kernel(freq,last_time,
+                                                freq,last_time,(1,0))
+        coup = self._bath.correlations.spectral_density(freq)*dw
+        bath_energy = np.diag(np.cumsum(np.cumsum(_sys_correlations.real*re_kernel+\
+                             1j*_sys_correlations.imag*im_kernel,axis=0),axis=1)).real*coup
+        bath_energy = np.append([0],bath_energy)
+        return self._process_tensor.times, bath_energy
+
+    def correlation(self,
+                    freq_1: float,
+                    time_1: float,
+                    freq_2: Optional[float] = None,
+                    time_2: Optional[float] = None,
+                    dw: Optional[tuple] = (1.0,1.0),
+                    dagg: Optional[tuple] = (1,0)):
+        r"""
+        Function to calculate two-time correlation function between to frequency
+        bands of a bath.
+
+        Parameters
+        ----------
+        freq_1 : float
+            Frequency of the later time operator.
+        time_1 : float
+            Time the later operator acts.
+        freq_2 : float (default = None)
+            Frequency of the earlier time operator. If set to None will default
+            to freq_2=freq_1.
+        time_2 : float (default = None)
+            Time the earlier operator acts. If set to None will default to
+            time_2=time_1.
+        dw : tuple (default = (1.0,1.0)):
+            Bandwidth of about each frequency comparing correlations between.
+            By default what is returned by this method is a correlation
+            *density*.
+        dagg : Optional[tuple] (default = (1,0))
+            Determines whether each operator is daggered or not e.g. (1,0)
+            would correspond to < a^\dagger a >
+        Returns
+        -------
+        times : tuple
+            Pair of times of each operation.
+        correlation : complex
+            Bath correlation function
+            <a^{dagg[0]}_{freq_1} (time_1) a^{dagg[1]}_{freq_2} (time_2)>
+        """
+        dt = self._process_tensor.times[1]-self._process_tensor.times[0]
+        corr_mat_dim = int(np.round(time_1/dt))
+        current_corr_dim = self._system_correlations.shape[0]
+        if time_2 is None:
+            time_2 = time_1
+        assert time_2 <= time_1, \
+            "The argument time_1 must be greater than or equal to time_2"
+        if freq_2 is None:
+            freq_2 = freq_1
+        correlation_set = [(x,y) for y in range(current_corr_dim,corr_mat_dim)\
+                           for x in range(y+1)]
+        if len(correlation_set)>0:
+            dim_diff = corr_mat_dim-self._system_correlations.shape[0]
+            coup_op = self.bath.coupling_operator
+            _new_sys_correlations = \
+                self._process_tensor.calc_correlations(coup_op,correlation_set)
```

Comment:
This method to calculate two time correlations currently doesn't exist on the public repository. We'll add it in soon.